### ISSUE TITLE
Incorrect Data Fields

### ISSUE THREAD

**Issue by david-j-duran** [pbutterworth/nissan-leaf-obd-ble#4]
Suspect this is a difference in how the data is transmitted for different model years, trims, and/or battery size.

Mine is the basic 2023 Leaf S w/ 40kWh pack

Odometer. Seems to be treating the dash number as km and converting to miles. There is a setting in LSP that resolved this for me "CAN Odometer in Miles" enabled.
- My dash: 12,475 miles
- OBD integration: 7,758 miles
- 12,475 / 1.609 km/mi = 7,753

Miles Remaining: has been static at 9.4 miles while my real range is ~100 miles. Think this is pulling in the wrong field.

<img width="377" alt="image" src="https://github.com/user-attachments/assets/419fe114-327b-4a5f-a614-f10b9b4752f5">

**Comment by pb-asbuilt** [comment]
Yeah, range remaining has never worked for me either. I have a note in the code to sort this out (one day).

The odometer one is interesting. I didn't realise that HA had the capability to convert sensors to different units. AFAIK, there is no way of know whether the raw data is km or mi. Might need to add a user setting for that to the options config.

**Comment by david-j-duran** [comment]
For clarity on the odometer, I didn’t mean to imply it’s an HA unit conversion. See this [Reddit thread](https://www.reddit.com/r/leaf/comments/v988j6/comment/ibuv8x5/) or [forum post](https://mynissanleaf.com/threads/odometer-on-leafspy-pro.33265/).

I’m not 100% sure where the conversion happens but OBD seems to report my dash value divided by 1.6 for some reason.

I’ve just made a template sensor to convert it back again.

**Comment by david-j-duran** [comment]
Also for the remaining range, I’ve just created a template to multiple SOC by 150mi. Close enough for government work.

**Comment by pb-asbuilt** [comment]
@david-j-duran, would be interesting to see the raw data coming from the car:
Enable debug logging for a while (while the car is on)
<img width="540" alt="image" src="https://github.com/user-attachments/assets/39360cf0-f585-41a1-b75a-16b4c55ec5f2">
If you aren't familiar with debug logging, once you stop the logging, it will download the log file.
You'll be able to see the raw bytes in the log. Cross reference with this doc:
[Leaf2018-CAN.pdf](https://github.com/user-attachments/files/17180925/Leaf2018-CAN.pdf)
Bear in mind, that my 2021 Leaf has discrepancies compared with this. Pinch of salt needed.

**Comment by randyoo** [comment]
Just got the bluetooth proxy working, and it picked up my LEAF right away--thanks!

Came here to report the same observations.

My US model LEAF must be using miles as the native unit, so HA is converting the reported miles as if they were km (as per         line 141 of custom_components/nissan_leaf_obd_ble/sensor.py: native_unit_of_measurement="km"), yielding a figure about 62% of the actual reading. If different region LEAFs report different units, a user config setting would be a good approach.

The "range remaining" is also nonsensical for me, but not so important me. Would love to get odometer working right, though, as it would make it possible to get a kWh/mile figure for each trip.
